### PR TITLE
Wrap up project

### PR DESCRIPTION
**Pull request by sabinem**
This PR wraps the project up: 
- improve documentation
- improve consistency
- make project DRY

### CODE REVIEW

**Comment by sabinem** [comment]
@cmdoret best dev review this commit by commit. In order to prove read the updated readme it might be helpful to read the readme also in the branch: https://github.com/statistikZH/statbotData/tree/wrap-up-project

**Comment by sabinem** [comment]
@cmdoret: I have followed up on all your suggestions. I think the functions are save enough for having them in the scripts.

File: README.md [hunk 1]
```diff
@@ -1,84 +1,156 @@
...
+
+- it provides questions and queries as [Statbot Training Data](#statbot-training-data) for the [Statbot Chatbot](#statbot-chatbot):
+
+```mermaid
+ flowchart TD
+    C[Federal Datasets] -->|Preprocess| B(Statbot postgress DB)
+    A[Cantonal Datasets] -->|Preprocess| B(Statbot postgress DB)
+    A[Cantonal Datasets] -->|Preprocess| B(Statbot postgress DB)
+    C[Federal Datasets] -->|Preprocess| B(Statbot postgress DB)
+    D{Ask Questions} -->|Statbot Swiss| B(Statbot postgress DB)
```
Comment (cmdoret) [inline README.md:20]:
```suggestion
flowchart TD
    Fed[Federal Datasets] -->|Preprocess| Scripts[Statbot scripts]
    Scripts -->|Upload| DB(Statbot postgress DB)
    Can[Cantonal Datasets] -->|Preprocess| Scripts
    User -->|Ask Questions| Chatbot(Statbot Swiss)
    Chatbot -->|query| DB(Statbot postgress DB)
```
**suggestion:** Show the chatbot and database as separate in the visualization

```mermaid
flowchart TD
    Fed[Federal Datasets] -->|Preprocess| Scripts[Statbot scripts]
    Scripts -->|Upload| DB(Statbot postgress DB)
    Can[Cantonal Datasets] -->|Preprocess| Scripts

    User -->|Ask Questions| Chatbot(Statbot Swiss)
    Chatbot -->|query| DB(Statbot postgress DB)
```

File: README.md [hunk 2]
```diff
@@ -1,84 +1,156 @@
...
 
 The repo consists of the following parts:
-- **input data**: in `data/const/statbot_input_data.csv`: imported datasets and spatial units as csv files
-- **package `statbotData`**: the function in the directory `R` make up a package of reusable functions, that are used in the pipeline to import the datasets into postgres 
-- **app**: `app.R` that gives an overview about the repo and the pipelines
-- **pipelines**: in `pipelines`: they contain the dataset processing as it is preapared for the postgres import and the test data (question / SQL query pairs) for the ML model per dataset
+- **input datasets**: contains the list input datasets for the pipelines and also a list of Swiss adminstative units, that are linked to each dataset
+- **R package `statbotData`**: the functions in the directory `R` make up a R package of reusable functions, that are used throughout the pipelines and scripts 
+- **pipelines** per **dataset**: they contain scripts for downloading datasets from its origin and preprocessing it for the postgres integration into Statbot Postgres DB](#statbot-postgres-db). They also include other files that relate to the dataset such as the [Statbot Training Data].
```
Comment (cmdoret) [inline README.md:29]:
```suggestion
- **pipelines** per **dataset**: they contain scripts for downloading datasets from their origin and preprocessing them for the postgres integration into [Statbot Postgres DB](#statbot-postgres-db). They also include other files that relate to the dataset such as the [Statbot Training Data](#statbot-training-data).
```
**typo:** plural + markdown section formatting

File: README.md [hunk 3]
```diff
@@ -1,84 +1,156 @@
...
+    class PxCube{
+      px_id
+      access_url
+      size
+      encoding
+    }
+    class CsvFile{
+      download_url
+    }
+    class ExelSheet{
```
Comment (cmdoret) [inline README.md:67]:
```suggestion
    class ExcelSheet{
```
**typo:** excel

File: README.md [hunk 4]
```diff
@@ -1,84 +1,156 @@
...
-### Functions to use in the pipeline scripts
+```mermaid
+erDiagram
+    metadata_tables
+    metadata_table_columns
+    dataset_observation }o--|| spatial_unit : spatial_scope
+ ```
 
-#### Step 1: Creating the dataset class
+**dataset**: each dataset is uploaded into exactly one table in [Statbot Postgres DB](#statbot-postgres-db): all tables contain these two columns mandatory:
```
Comment (cmdoret) [inline README.md:109]:
```suggestion
**dataset**: each dataset is uploaded into exactly one table in [Statbot Postgres DB](#statbot-postgres-db): all tables contain these two mandatory columns:
```
**typo:** grammar

File: README.md [hunk 5]
```diff
@@ -94,72 +166,117 @@ JOIN spatial_unit as S on T.spatialunit_uid = S.spatialunit_uid
...
 
-Update the date in the metadata if you rerun the pipeline if needed:
+The `queries.sql` files contain the [Statbot Training Data](#statbot-training-data) for each dataset. See this section for a detailed description of these files.
 
-```
-statbotData::update_pipeline_last_run_date(ds)
-```
+#### `queries.log`
+
+These files are the output of a run of `statbotData::testrun_queries(ds)` on the queries in the `queries.sql` file of the dataset. A datatime indicates the datetime of the run. It contains also information whether the file was run on the remote postgres instance  `REMOTE` or on the local `ds$postgres` data frame.
```
Comment (cmdoret) [inline README.md:201]:
```suggestion
These files are the output of a run of `statbotData::testrun_queries(ds)` on the queries in the `queries.sql` file of the dataset. A timestamp indicates the date and time of the run. It also contains information on whether the file was run on the remote postgres instance  `REMOTE` or on the local `ds$postgres` data frame.
```
**typo:** dat**a**time (suggested timestamp instead)

File: README.md [hunk 6]
```diff
@@ -94,72 +166,117 @@ JOIN spatial_unit as S on T.spatialunit_uid = S.spatialunit_uid
...
+#### `metadata_table_columns_.csv`
+
+These are metadata for the table columns in postgres. They are loaded into the postgres metadata table `metadata_table_columns`, see [Statbot Postgres DB](#statbot-postgres-db) for details. 
 
-Then upload the metadata to postgres: 
+## Scripts
 
+Originally the repo did not have a scripts directory and all the work like test running the queries and uploading metadata and data to postgres had been done directly in the pipelines. 
 
+But the code for these choires is always the same and consists of using the appropriate functions of the package `statbotData`. It does not depend on the specific pipeline or dataset, but on the situation and the status of the dataset. Therefore this code has been extracted from the pipelines and put into a directory of its own with a selection of different sample scripts that relate to certain scenarios:
```
Comment (cmdoret) [inline README.md:215]:
```suggestion
But the code for these chores is always the same and consists of using the appropriate functions of the package `statbotData`. It does not depend on the specific pipeline or dataset, but on the situation and the status of the dataset. Therefore this code has been extracted from the pipelines and put into a directory of its own with a selection of different sample scripts that relate to certain scenarios: 
```
**typo:** chores(?)

File: README.md [hunk 7]
```diff
@@ -94,72 +166,117 @@ JOIN spatial_unit as S on T.spatialunit_uid = S.spatialunit_uid
...
+These are metadata for a table in postgres. They are loaded into the postgres metadata table `metadata_tables`, see section on the [Statbot Postgres DB](#statbot-postgres-db) above. 
+
+#### `metadata_table_columns_.csv`
+
+These are metadata for the table columns in postgres. They are loaded into the postgres metadata table `metadata_table_columns`, see [Statbot Postgres DB](#statbot-postgres-db) for details. 
 
-Then upload the metadata to postgres: 
+## Scripts
 
+Originally the repo did not have a scripts directory and all the work like test running the queries and uploading metadata and data to postgres had been done directly in the pipelines.
```
Comment (cmdoret) [inline README.md:213]:
```suggestion
```
**nitpick(non-blocking):** I think we could remove this to focus on the current status.

**Review by cmdoret — CHANGES_REQUESTED** [review]
**praise:** The readme looks great! I found typos and had a few suggestions/

**Review by cmdoret — APPROVED** [review]
:raised_hands: All good for me!

**Pull request merged**